plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -170,40 +170,72 @@ func PrioritizeNodes(pod *api.Pod, podLister algorithm.PodLister, priorityConfig
 		return EqualPriority(pod, podLister, nodeLister)
 	}
 
+	var (
+		mu  sync.Mutex
+		err error
+	)
 	combinedScores := map[string]int{}
+	errC := make(chan error)
+
 	for _, priorityConfig := range priorityConfigs {
-		weight := priorityConfig.Weight
 		// skip the priority function if the weight is specified as 0
-		if weight == 0 {
+		if priorityConfig.Weight == 0 {
 			continue
 		}
-		priorityFunc := priorityConfig.Function
-		prioritizedList, err := priorityFunc(pod, podLister, nodeLister)
-		if err != nil {
-			return schedulerapi.HostPriorityList{}, err
-		}
-		for _, hostEntry := range prioritizedList {
-			combinedScores[hostEntry.Host] += hostEntry.Score * weight
+		go func(algorithm.PriorityConfig) {
+			weight := priorityConfig.Weight
+			priorityFunc := priorityConfig.Function
+			prioritizedList, err := priorityFunc(pod, podLister, nodeLister)
+			if err != nil {
+				errC <- err
+				return
+			}
+			for _, hostEntry := range prioritizedList {
+				mu.Lock()
+				combinedScores[hostEntry.Host] += hostEntry.Score * weight
+				mu.Unlock()
+			}
+			errC <- nil
+		}(priorityConfig)
+	}
+	// wait for all go routines to finish
+	for i := 0; i < len(priorityConfigs); i++ {
+		nerr := <-errC
+		if err == nil && nerr != nil {
+			err = nerr
 		}
 	}
+	if err != nil {
+		return schedulerapi.HostPriorityList{}, err
+	}
+
+	var wg sync.WaitGroup
 	if len(extenders) != 0 && nodeLister != nil {
 		nodes, err := nodeLister.List()
 		if err != nil {
 			return schedulerapi.HostPriorityList{}, err
 		}
-
 		for _, extender := range extenders {
-			prioritizedList, weight, err := extender.Prioritize(pod, &nodes)
-			if err != nil {
-				// Prioritization errors from extender can be ignored, let k8s/other extenders determine the priorities
-				continue
-			}
+			wg.Add(1)
+			go func(algorithm.SchedulerExtender) {
+				defer wg.Done()
+				prioritizedList, weight, err := extender.Prioritize(pod, &nodes)
+				if err != nil {
+					// Prioritization errors from extender can be ignored, let k8s/other extenders determine the priorities
+					return
+				}
 
-			for _, hostEntry := range *prioritizedList {
-				combinedScores[hostEntry.Host] += hostEntry.Score * weight
-			}
+				for _, hostEntry := range *prioritizedList {
+					mu.Lock()
+					combinedScores[hostEntry.Host] += hostEntry.Score * weight
+					mu.Unlock()
+				}
+			}(extender)
 		}
 	}
+	// wait for all go routines to finish
+	wg.Wait()
+
```

Comment:
I would rather keep the previous behavior. We can revisit this once there is a real issue. (I am not sure about this since we do not have any test with extender, and we probably do not want to do optimization for extender.)